scipy/linalg/_matfuncs_expm.c

```diff
@@ -2179,8 +2179,8 @@ pade_UV_calc_z(EXPM_Z* Am, const Py_ssize_t size_n, const int m, int* info)
                 inter1 = _Cmulcr(Am[n2 + n*i + j], b[2]);
                 inter2 = _Cmulcr(Am[2*n2 + n*i + j], b[4]);
                 inter3 = _Cmulcr(Am[3*n2 + n*i + j], b[6]);
-                Am[4*n2 + n*i + j] = CPLX_Z(creal(inter1) + creal(inter2) + creal(inter3),
-                                            cimag(inter1) + cimag(inter2) + cimag(inter3));
+                Am[n2 + n*i + j] = CPLX_Z(creal(inter1) + creal(inter2) + creal(inter3),
```

Comment:
Can't help wondering if there's a generic solution for these kinds of index/pointer typos. 

(idle wonderings, nothing blocking the PR itself)

Reply:
A better language would be really nice, or dropping support for non-compliant compilers. I should have tested on Conda but wasn't able to for quite a while. Now with flang-new I managed to reproduce the issue.